src/main/java/com/sintern/api/OpenInternPositionController.java

```diff
@@ -1,37 +1,66 @@
 package com.sintern.api;
 
+import com.sintern.domain.dto.OpenInternPositionDTO;
+import com.sintern.domain.entity.Company;
+import com.sintern.domain.entity.CompanyLogo;
 import com.sintern.domain.entity.Domain;
+import com.sintern.domain.entity.OpenInternPosition;
 import com.sintern.domain.enums.DomainType;
-import com.sintern.domain.dto.OpenInternPositionDTO;
-import com.sintern.service.OpenInternPositionService;
+import com.sintern.service.LogoService;
+import com.sintern.service.OpenInternPositionServiceImpl;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
 
 @CrossOrigin
 @RestController
 @RequestMapping("/open-positions")
 public class OpenInternPositionController {
-    private final OpenInternPositionService openInternPositionService;
+    private final OpenInternPositionServiceImpl openInternPositionService;
+    private final LogoService logoService;
 
     @Autowired
-    public OpenInternPositionController(OpenInternPositionService openInternPositionService) {
+    public OpenInternPositionController(OpenInternPositionServiceImpl openInternPositionService, LogoService logoService) {
         this.openInternPositionService = openInternPositionService;
+        this.logoService = logoService;
     }
 
     @RequestMapping(value = "/address/{address}", method = RequestMethod.GET)
-    public List<OpenInternPositionDTO> filterByAddress(@PathVariable String address){
+    public List<OpenInternPositionDTO> filterByAddress(@PathVariable String address) {
         return openInternPositionService.findOpenInternPositionDTOByCompanyAddress(address);
     }
 
     @RequestMapping(value = "/domain/{domainType}", method = RequestMethod.GET)
-    public List<OpenInternPositionDTO> filterByDomain(@PathVariable String domainType){
+    public List<OpenInternPositionDTO> filterByDomain(@PathVariable String domainType) {
         return openInternPositionService.findOpenInternPositionDTOByCompanyDomain(DomainType.valueOf(domainType));
     }
 
     @RequestMapping(value = "/domains", method = RequestMethod.GET)
-    public List<Domain> getDomains(){
+    public List<Domain> getDomains() {
         return openInternPositionService.getDomains();
     }
+
+    @GetMapping("{openPositionId}/company-logo/")
+    public ResponseEntity<byte[]> getLogo(@PathVariable UUID openPositionId) {
+
+        Optional<CompanyLogo> companyLogoOptional = logoService.getLogoByCompanyId(openInternPositionService.getOpenInternPositionById(openPositionId).getCompany().getId());
```

Comment:
This is quite hard to read. Consider extracting `openInternPositionService.getOpenInternPositionById(openPositionId).getCompany()` to a variable.